### ISSUE TITLE
Some tests failing in parallel

### ISSUE THREAD

**Issue by crswong888** [idaholab/falcon#46]
I'm not sure I understand the physics well enough to try and fix it myself, but I have identified 4 tests that lead to an EXODIFF failure when running with MPI:

```
PT_T_convection.PT_T_convection ........................................................ FAILED (EXODIFF)
PT_T_convection.PT_T_convection_SUPG ................................................... FAILED (EXODIFF)
PT_T_convection.PT_T_convection_SUPG_gaussian .......................................... FAILED (EXODIFF)
PT_T_SideProductionIntegral.PT_T_SideProductionIntegral ................................ FAILED (EXODIFF)
```

These tests pass in serial, i.e., when running with `./run_tests -j4`, but not if you run with `./run_tests -j4 -p2`, which is equivalent to executing each individual test with the `mpiexec -n 2` command. 

An additional test fails when multithreading. Running `./run_tests -j4 -p2 --n-threads=2` leads to the following results:

```
PT_T_convection.PT_T_convection ........................................................ FAILED (EXODIFF)
PT_T_convection.PT_T_convection_gaussian ............................................... FAILED (EXODIFF)
PT_T_convection.PT_T_convection_SUPG ................................................... FAILED (EXODIFF)
PT_T_convection.PT_T_convection_SUPG_gaussian .......................................... FAILED (EXODIFF)
PT_T_SideProductionIntegral.PT_T_SideProductionIntegral ................................ FAILED (EXODIFF)
```

**Comment by rpodgorney** [comment]
I'm not getting these failures.......

Let me dig in a bit

**Comment by crswong888** [comment]
@rpodgorney I am getting them on both my Linux desktop and my macbook, which I know for sure handle floating-point roundoffs a bit differently. You ought to try running on a different machine if you have one or updating your libmesh or conda environment if you haven't done so recently.

**Comment by rpodgorney** [comment]
@crswong888 -- I see the failed tests, I don't think I have ever ran the full suite in parallel.  I'll look at the failed tests closer over the next day or so

**Comment by crswong888** [comment]
@rpodgorney yeah the `-p` argument for `run_tests` seems to obscured. I only just discovered it recently while working on a libMesh PR that I thought I was finally done with, only to realize that I was only halfway done because it didn't work in parallel as I thought that's what `-j` was for.

I'm seeing all tests pass tests with `-p` now. But I am still getting the `test:PT_T_convection.PT_T_convection_gaussian` test failure when using both parallel and multithreading, i.e., `./run_tests -p2 --n-threads=2`. It's the only one now. Interestingly, if I only use threading like `./run_tests --n-threads=2`, I don't get the error.

**Comment by rpodgorney** [comment]
@crswong888 it must have something to do with mesh partitioning.  Stat another issue and I'l work on it.

Did it fail on all OSs?

**Comment by crswong888** [comment]
@rpodgorney yeah I'm getting the same behavior on MacOS and Ubuntu. I might spend a few minutes trying to fix it myself because I see now what you did with the other tests, and maybe something similar can be done. If it proves to be difficult, I'll open a new issue.
